src/lib/plugins/world.js

```diff
@@ -243,6 +253,12 @@ module.exports.player = function (player, serv, settings) {
       .catch((err) => setTimeout(() => { throw err }), 0)
   }
 
+  // todo as I understand need to handle difficulty packet instead?
+  player.onPlayerChangeRenderDistance = (newDistance = player.view, forced = false) => {
```

Comment:
Why this function instead of listening to an event ?

Reply:
Changed to event. I dont mind having it as event.